pkg/api/v1/clients/apiclient/resource_client_test.go

```diff
@@ -59,7 +59,11 @@ var _ = Describe("Apiclient", func() {
 
 		port = lis.Addr().(*net.TCPAddr).Port
 		fmt.Fprintf(GinkgoWriter, "grpc listening on %v\n", port)
-		go server.Serve(lis)
+		go func() {
+			defer GinkgoRecover()
```

Comment:
I haven't seen this pattern, yet.  Can you help me understand what it's doing?

Reply:
Sure! https://onsi.github.io/ginkgo/#mental-model-how-ginkgo-handles-failure has more details.

```
However, if your spec launches a goroutine that calls Fail (or, equivalently, invokes a failing Gomega assertion), there's no way for Ginkgo to rescue the panic that Fail throws. This will cause the suite to panic and no subsequent specs will run. To get around this you must rescue the panic using defer GinkgoRecover()
```

This change likely wasn't required (as it has lived in our codebase for a while). But I think it's good practice for any goroutines spawned within a test to include the GinkgoRecover.